### ISSUE TITLE
CircleCI job fails to establish connection with SauceLabs periodically

### ISSUE THREAD

**Issue by AndrewKushnir** [angular/angular#28170]
The `legacy-unit-tests-saucelabs` CircleCI job is failing periodically with the following error:

```
14 Jan 18:38:08 - couldn't connect to https://saucelabs.com/rest/v1/angular-ci/tunnels: Post https://saucelabs.com/rest/v1/angular-ci/tunnels: net/http: timeout awaiting response headers
14 Jan 18:38:08 - Sauce Connect could not establish a connection.
14 Jan 18:38:08 - Please check your firewall and proxy settings.
14 Jan 18:38:08 - You can also use the sc --doctor to launch Sauce Connect in diagnostic mode.
14 Jan 18:38:08 - Goodbye.
```

The problem happens more often when we merge several PRs in a short period of time (rate limiting?). It'd be great to investigate what causing this, so we have less flakiness while running tests.

cc @IgorMinar @devversion

**Comment by devversion** [comment]
@AndrewKushnir It could be related to rate limiting, though this usually results in a different error message. 

I've been chatting with the Saucelabs support some time ago and they've said that we sometimes hit the API rate limit. 

The rate limit is 3500 requests within an hour. I somehow doubt that we can often hit this number by only making `1 + {X: launched_browsers}` requests for a PR. One request for the tunnel, and X requests for reporting test results back to Saucelabs.

I confirmed with Saucelabs that requests to launch a browser do not count to the rate limit.

-- 

Though this error looks overall related to the amount of requests we make to Saucelabs. Maybe since we made so many requests in a short period of time, Saucelabs didn't accept any connection at all (instead of returning a proper error response)
